# HG changeset patch
# User Nathan Carter <ncarter@bentley.edu>
# Date 1326129896 18000
# Node ID 066395b9c394a33e85d00df65b8c28f6e34a7213
# Parent  269237bdd1f5969ef933e9667f601fcbbd78b2dc
Updating previous patch to newer Sage build

diff --git a/sage/plot/plot_field.py b/sage/plot/plot_field.py
--- a/sage/plot/plot_field.py
+++ b/sage/plot/plot_field.py
@@ -218,10 +218,21 @@
 
         sage: x,y = var('x y')
         sage: plot_slope_field(sin(x+y)+cos(x+y), (x,-3,3), (y,-3,3))
+
+    Plot a slope field using a lambda function::
+
+        sage: plot_slope_field(lambda x,y: x+y, (-2,2), (-2,2))
     """
     slope_options = {'headaxislength': 0, 'headlength': 0, 'pivot': 'middle'}
     slope_options.update(kwds)
 
     from sage.functions.all import sqrt
-    norm = sqrt((f**2+1))
-    return plot_vector_field((1/norm, f/norm), xrange, yrange, **slope_options)
+    from inspect import isfunction
+    if isfunction(f):
+        norm_inverse=lambda x,y: 1/sqrt(f(x,y)**2+1)
+        f_normalized=lambda x,y: f(x,y)*norm_inverse(x,y)
+    else:
+        norm_inverse = 1/sqrt((f**2+1))
+        f_normalized=f*norm_inverse
+    return plot_vector_field((norm_inverse, f_normalized), xrange, yrange, **slope_options)
+
